### ISSUE TITLE
className prop does not propagate down to the HTML element

### ISSUE THREAD

**Issue by empz** [tcampb/react-calendly#56]
I'm using Tailwind and I need to apply some classes to the resulting `<a>` element. 

The current implementation only propagates the `style` prop which is not really compatible with Tailwind.

**Comment by empz** [comment]
I've created a PR for this: https://github.com/tcampb/react-calendly/pull/57

**Comment by tcampb** [comment]
Hi @empz,

Thanks for creating a PR to address this issue! I agree that this should render a button tag instead of an anchor tag; I decided to keep it as an anchor tag so that it would match Calendly's popup text code:

```html
<!-- Calendly link widget begin -->
<link href="https://assets.calendly.com/assets/external/widget.css" rel="stylesheet">
<script src="https://assets.calendly.com/assets/external/widget.js" type="text/javascript" async></script>
<a href="" onclick="Calendly.initPopupWidget({url: 'https://calendly.com/acmesales'});return false;">Schedule time with me</a>
<!-- Calendly link widget end -->
```

Could you use the [`openPopupWidget`](https://github.com/tcampb/react-calendly#how-do-i-create-a-custom-button-that-triggers-a-pop-up-scheduler) function to create a custom button instead of using the PopupText component?

**Comment by empz** [comment]
I see, but I guess this library can do better than Calendly's original snippet ;)
In the end, all it matters is the calls to `Calendly`'s methods.

Thanks, I'll use `openPopupWidget` with my own button!

**Comment by tcampb** [comment]
> I see, but I guess this library can do better than Calendly's original snippet ;)
> In the end, all it matters is the calls to `Calendly`'s methods.
> 
> Thanks, I'll use `openPopupWidget` with my own button!

Good point! What are your thoughts on adding `role="button"` to the anchor tag in PR https://github.com/tcampb/react-calendly/pull/57? Again, thanks for bringing this issue up and creating a PR!

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by empz** [comment]
I guess it's better than nothing. I still think a native button should be used instead of an anchor. But yeah, I'll add role="button" to the PR.

**Comment by tcampb** [comment]
@empz - Good point, would you mind updating this to a button in your PR? If this is too large of a change then I'm happy to create a separate PR, just let me know!

Thanks!

**Comment by empz** [comment]
@tcampb Sure, but would you want to style it as an anchor by default? Or should we leave it as a button and leave the styling to the user?

**Comment by tcampb** [comment]
If it's not too much additional effort then I'd prefer to set the default styles to match an anchor link by default so that it's not a "breaking" change. For example:

```typescript
    return (
      <button onClick={createClickHandler(widgetOptions)} style={this.props.styles || ANCHOR_LINK_STYLES}>
        {this.props.text}
      </button>
    );
 ```
 
That said, if it becomes difficult for developers to override the default styles then I'm okay with just creating a regular button. Do you have a preference for your use-case?

**Comment by empz** [comment]
The problem is that it's impossible to style a button as an anchor with only inline styles (can't do :hover: underline). I'm not familiar with the setup of the project, is there a way to import a css?

**Comment by empz** [comment]
In my opinion, I'd prefer a floating popup to be opened by clicking a button rather than a link. A link should navigate.

I think if we propagate both className and styles to the button, the user can style it however they want. It's the most flexible.

**Comment by tcampb** [comment]
That makes sense, a button without default styles sounds good to me. Could you also rename this component to `PopupButton`? I'll update the package's major version before publishing the changes.

**Comment by empz** [comment]
Done.

I guess the example of using openPopupWidget is kind of obsolete now. But
the function itself is still useful to trigger the popup programmatically.

**Comment by tcampb** [comment]
Thanks @empz! Your PR changes are included in [version 2.0.0](https://www.npmjs.com/package/react-calendly/v/2.0.0).

**Comment by Shikhar2404** [comment]
can i change the amount of money dynamically from webview to calendly? or is there any way i can send calculate price from my RN and send to calendly's event. i want to give discount to users
